src/bench/rpc_blockchain.cpp

```diff
@@ -40,7 +40,7 @@ static void BlockToJsonVerbose(benchmark::Bench& bench)
 {
     TestBlockAndIndex data;
     bench.run([&] {
-        auto univalue = blockToJSON(data.block, &data.blockindex, &data.blockindex, /*verbose*/ true);
+        auto univalue = blockToJSON(WITH_LOCK(::cs_main, return std::ref(data.testing_setup->m_node.chainman->m_blockman)), data.block, &data.blockindex, &data.blockindex, /*verbose*/ true);
```

Comment:
In commit "Move fHavePruned to BlockManager" (b40f35d7f0507fcfb71d1c950c5158d6ff5fdb5e)

This is another place like https://github.com/bitcoin/bitcoin/pull/22564#discussion_r697594297 where it looks like m_blockman is improperly annotated to require cs_main. In general, if you have a variable

```c++
T variable GUARDED_BY(mutex);
```

And you are writing `WITH_LOCK(mutex, return &variable);` you're just bypassing the annotation, and either the annotation is wrong and shouldn't be there, or it's right and you're introducing a possible race condition when the variable is accessed through the pointer later without the lock held.

It would be nice to drop the `m_blockman` annotation in this PR because it seems overly broad. Or if it's easier a comment could be added just to say why the annotation is bypassed here.